src/python/pants/core/goals/update_build_files_test.py

```diff
@@ -184,6 +187,58 @@ def test_black_config(black_rule_runner: RuleRunner) -> None:
     assert Path(black_rule_runner.build_root, "BUILD").read_text() == "tgt(name='t')\n"
 
 
+# ------------------------------------------------------------------------------------------
+# Yapf formatter fixer
+# ------------------------------------------------------------------------------------------
+
+
+@pytest.fixture
+def yapf_rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=(
+            format_build_file_with_yapf,
+            update_build_files,
+            *pants_bin.rules(),
+            *config_files.rules(),
+            *pex.rules(),
+            SubsystemRule(Yapf),
+            SubsystemRule(UpdateBuildFilesSubsystem),
+            UnionRule(RewrittenBuildFileRequest, FormatWithYapfRequest),
+        )
+    )
+
+
+def test_yapf_fixer_fixes(yapf_rule_runner: RuleRunner) -> None:
+    yapf_rule_runner.write_files({"BUILD": "tgt( name =  't' )"})
+    result = yapf_rule_runner.run_goal_rule(UpdateBuildFilesGoal, env_inherit=BLACK_ENV_INHERIT)
+    assert result.exit_code == 0
+    assert result.stdout == dedent(
+        """\
+        Updated BUILD:
+          - Format with Yapf
+        """
+    )
+    assert Path(yapf_rule_runner.build_root, "BUILD").read_text() == 'tgt(name="t")\n'
+
+
+def test_yapf_fixer_noops(yapf_rule_runner: RuleRunner) -> None:
+    yapf_rule_runner.write_files({"BUILD": 'tgt(name="t")\n'})
+    result = yapf_rule_runner.run_goal_rule(UpdateBuildFilesGoal, env_inherit=BLACK_ENV_INHERIT)
+    assert result.exit_code == 0
+    assert Path(yapf_rule_runner.build_root, "BUILD").read_text() == 'tgt(name="t")\n'
+
+
+def test_yapf_fixer_args(yapf_rule_runner: RuleRunner) -> None:
+    yapf_rule_runner.write_files({"BUILD": "tgt(name='t')\n"})
+    result = yapf_rule_runner.run_goal_rule(
+        UpdateBuildFilesGoal,
+        global_args=["--yapf-args=--style='{indent_width: 2}'"],
```

Comment:
It looks like this won't cause the BUILD file to be rewritten. Instead, you'd want to use the original BUILD file with 4-space indentation and check that it becomes 2 spaces.